R/Rules-methods.R

```diff
@@ -663,6 +663,35 @@ setMethod("nextBest",
 
 ## ============================================================
 
+##' @describeIn nextBest Method for `NextBestInfTh` class, which will give
+##'   the appropriate dose within an information theoretic framework
+##' @export
+setMethod("nextBest", signature = signature(nextBest = "NextBestInfTh", doselimit = "numeric", samples = "Samples",
+                                            model = "Model", data = "Data"),
+          def = function(nextBest, doselimit, samples, model, data, ...){
+
+            probSamples <- matrix(nrow=sampleSize(samples@options),
+                                  ncol=data@nGrid)
+
+            for(i in seq_len(data@nGrid))
+            {
+              probSamples[, i] <- prob(dose=data@doseGrid[i],
+                                       model,
+                                       samples)
+            }
+
+            dosesOK <-
+              if(length(doselimit))
+                (data@doseGrid <= doselimit)
+            else
+              rep(TRUE, length(data@doseGrid))
+
+           criterion <-colMeans(((probSamples-nextBest@target)^2)/(probSamples^nextBest@asymmetry*(1-probSamples)^(2-nextBest@asymmetry)))
```

Comment:
maybe I would suggest here to have a little helper function so that here you can just call 
```suggestion
           criterion <- colMeans(h_info_theory_dist(probSamples, nextBest@target, nextBest@asymmetry))
```